File: tests/resources/EmptyMappingController.java
package com.example.controller;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/users")
public class EmptyMappingController {

    @GetMapping("")
    public String getUsersWithEmptyMapping() {
        return "users";
    }

    @PostMapping("")
    public String createUserWithEmptyMapping() {
        return "created";
    }

    @GetMapping("/details")
    public String getUserDetails() {
        return "details";
    }
}

File: tests/resources/EmptyMappingController.kt
package com.example.controller

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/products")
class EmptyMappingController {

    @GetMapping("")
    fun getProductsWithEmptyMapping(): String {
        return "products"
    }

    @PostMapping("")
    fun createProductWithEmptyMapping(): String {
        return "created"
    }

    @GetMapping("/details")
    fun getProductDetails(): String {
        return "details"
    }
}